src/main/java/cz/cvut/kbss/termit/service/repository/VocabularyRepositoryService.java

```diff
@@ -39,28 +40,35 @@
 
     private final ChangeRecordService changeRecordService;
 
-    final VocabularyImportService importService;
-
-    final ResourceRepositoryService resourceService;
+    private final ResourceRepositoryService resourceService;
 
     private final Configuration.Namespace config;
 
+    private final ApplicationContext context;
+
     @Autowired
-    public VocabularyRepositoryService(VocabularyDao vocabularyDao, IdentifierResolver idResolver,
+    public VocabularyRepositoryService(ApplicationContext context, VocabularyDao vocabularyDao, IdentifierResolver idResolver,
                                        Validator validator, ChangeRecordService changeRecordService,
-                                       @Lazy TermService termService, VocabularyImportService importService,
+                                       @Lazy TermService termService,
                                        @Lazy ResourceRepositoryService resourceService,
                                        final Configuration config) {
         super(validator);
+        this.context = context;
         this.vocabularyDao = vocabularyDao;
         this.idResolver = idResolver;
         this.termService = termService;
         this.changeRecordService = changeRecordService;
-        this.importService = importService;
         this.resourceService = resourceService;
         this.config = config.getNamespace();
     }
 
+    /**
+     * This method ensures new instances of the prototype-scoped bean are returned on every call.
+     */
+    private SKOSImporter getSKOSImporter() {
```

Comment:
Consider using `@Lookup` for injecting prototype beans. See `TermOccurrenceResolvers`